components/sidebar/sidebar_channel/sidebar_channel_icon/sidebar_channel_icon.tsx

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import ArchiveIcon from 'components/widgets/icons/archive_icon';
+import DraftIcon from 'components/widgets/icons/draft_icon';
+
+type Props = {
+    icon?: JSX.Element;
+    channelIsArchived: boolean;
+    hasDraft: boolean;
+};
+
+export default class SidebarChannelIcon extends React.PureComponent<Props> {
```

Comment:
Also, we could consider making this into a functional component since it's relatively simple

Reply:
Agreed, I plan to do a passover of all the components and turn as many into FCs as possible.
But I'd prefer to do that after the fact rather than change it to an FC only to find that it requires more logic.